docs/advanced.md

```diff
@@ -619,3 +619,28 @@ If you do need to make HTTPS connections to a local server, for example to test
 >>> r
 Response <200 OK>
 ```
+
+## Custom Transports
+
+HTTPX's `Client` also accepts a `transport` argument. This argument allows you
+to inject a custom Transport objects that will be used to perform the actual
+sending of the requests.
+
+These Transport objects must implement some methods from
+[`httpcore`'s API](https://www.encode.io/httpcore/api/), either
+`httpcore.AsyncHTTPTransport` if you're using the asynchronous `Client`, or
+`httpcore.SyncHTTPTransport` if you're using a synchronous one.
+
+Specifically you *MUST* implement `request`, and `close` or `aclose` depending
+on the type of client you're using.
+
+For example, HTTPX ships with a transport that uses the excellent
+[`urllib3` library](https://urllib3.readthedocs.io/en/latest/):
+
+```python
+>>> import httpx
+>>> from httpx._dispatch.urllib3 import URLLib3Dispatcher
```

Comment:
We'll want to rename this to `URLLib3Transport`, and promote it to public API, so `httpx.URLLib3Transport(...)`.

Reply:
I guess we also ought to note that `urllib3` needs to be installed to use this transport.

We *might* need to look at if we need an `assert` check in the codebase too, since `urllib3` is no longer going to be a core dependancy of `httpx`, it'll be an optional from 13.0 onwards.

Reply:
Yep, that makes sense 👍